### PR TITLE
fix(smart-object): return None for blank filetype, add detected_filetype

### PR DESCRIPTION
**Pull request by kyamagu**
Fixes #599

## Problem

Some tools (e.g. Adobe Illustrator) write four spaces (`b'    '`) as the linked-layer `filetype` field instead of a real 4-byte type code. `SmartObject.filetype` was silently returning `""`, which callers treated as unknown/`None` and had to work around manually.

### `SmartObject.filetype` → `str | None`
Returns `None` (not `""`) when the stored field is blank. This is a **minor breaking change**: code checking `if so.filetype:` continues to work; code doing `so.filetype == ""` will need updating.

### New `SmartObject.detected_filetype` property
Best-effort heuristic falling back through three levels:
1. Stored `filetype` (if non-`None`)
2. File extension from `filename` (lower-cased, null bytes stripped, trailing-dot edge case handled)
3. Magic-byte sniffing of the first embedded bytes — **data-kind only**, no filesystem access for external objects

Covers: PSD (v1), PSB (v2), PDF, PNG, JPEG, GIF, TIFF (LE/BE), WebP, ZIP, BMP. SVG intentionally excluded (too weak to detect reliably via magic bytes alone).

### `is_psd()` — magic-byte fallback, no extension
Falls back to `data[:4] == b"8BPS"` when `filetype` is `None`. Filename extension is **not** used to avoid misclassifying non-PSD files named `*.psd`.

### `__repr__` — shows both `type=` and `detected=`
Easier debugging when the stored type is blank.

### Docs
Updated stale `filetype` examples in `docs/usage.rst` and `SmartObjectLayer` docstring to use `detected_filetype`.

## Tests
30 new tests in `TestFiletypeBlank` covering:
- `filetype` returns `None` vs. stored value
- All magic byte types
- Filename extension fallback (including `foo.` trailing-dot edge case)
- External kind: confirmed no filesystem read in `detected_filetype`
- `is_psd()` via stored type, via magic, and negative cases (PDF named `*.psd`)
- `__repr__` with blank and present stored type

All 44 tests pass.

## Migration

```python
# Before (workaround needed)
if so.filetype:
    ext = so.filetype
else:
    ext = so.data[:5] == b'%PDF-' and 'pdf'  # manual sniffing

# After
ext = so.detected_filetype  # handles all cases
```

### CODE REVIEW

File: src/psd_tools/api/smart_object.py [hunk 1]
```diff
@@ -163,15 +191,71 @@ def filesize(self) -> int:
...
+        Returns ``None`` when the field is absent or blank (e.g. some
+        Adobe Illustrator smart objects store four spaces instead of a real
+        type code).  Use :py:attr:`detected_filetype` for a best-effort
+        guess that falls back to the filename extension and magic bytes.
+        """
         if self._data is None:
             raise ValueError("Smart object data not found")
-        return self._data.filetype.lower().strip().decode("ascii")
+        raw = self._data.filetype.lower().strip().decode("ascii")
+        return raw if raw else None
```
[the remark this answers is not in the document]
Comment (kyamagu) [reply src/psd_tools/api/smart_object.py]: Fixed in 7a4f5c6: changed the strip to `strip(b" \x00")` on the raw bytes before decoding, so both four-spaces and all-NUL fields are treated as blank and return `None`.

File: tests/psd_tools/api/test_smart_object.py [hunk 2]
```diff
@@ -241,3 +248,181 @@ def test_fullpath_outside_external_dir_falls_back_to_relpath(
...
+# filetype / detected_filetype / is_psd tests — Issue #599
+# ---------------------------------------------------------------------------
+
+
+class TestFiletypeBlank:
+    """filetype returns None for blank stored field; detected_filetype falls back."""
+
+    def test_filetype_returns_none_when_blank(self) -> None:
+        so = _make_data_smart_object("foo.pdf", b"%PDF-1.5 rest", filetype=b"    ")
+        assert so.filetype is None
```
[the remark this answers is not in the document]
Comment (kyamagu) [reply tests/psd_tools/api/test_smart_object.py:263]: Added two tests in 7a4f5c6: `test_filetype_returns_none_when_all_nul` and `test_detected_falls_back_when_filetype_all_nul`, covering the `b"\x00\x00\x00\x00"` case for both `filetype` and the `detected_filetype` fallback chain.
